### DISCUSSION TITLE
Stream Download Filenames

### DISCUSSION THREAD

**Discussion by eswiig** [stashapp/stash#2922] [Q&A] [answered]
Love Stash, 

But when downloading the stream, it autosaves as stream.mp4, is there any way to make it use the filename, scene name or even the path name (minus the folder path).

Just a bit tedious to having to manually rename it.

example: [Studio] Alice In Wonderland (2010-01-01).mp4 instead of stream.mp4

**Comment by ALonelyJuicebox** [comment]
Just trying to understand your use case-- are you trying to re-download your own content from your own Stash? The filepath should be under the File Info tab on the Scene page.

**Reply by eswiig** [reply]
Me and my friend each run our own stash, I want to download from his stash and I'd prefer if the browser prompted me to save the stream download as [Studio] Alice In Wonderland (2010-01-01).mp4 instead of stream.mp4 which I then have to go and rename manually.

To download a scene, click edit, then the url under Stream:

It just feels a bit tedious and unnecessary to have to manually rename from stream.mp4. 

Normally the browser will serve you a proper filename when you download from websites.

**Comment by feederbox826** [comment]
https://github.com/stashapp/stash/issues/627#issuecomment-672557118

**Comment by feederbox826** [comment, marked as answer]
Implemented in https://github.com/stashapp/stash/pull/6119
